### PR TITLE
drivers: gpio: Update NXP GPIO driver for the updated IP Block

### PR DESCRIPTION
**Pull request by mmahadevan108**
1. Move the GPIO mux setting to the soc layer. The GPIO MUX value may vary based on the SoC Family
2. Enable the digital input buffer if available

### CODE REVIEW

**Review by danieldegrasse — APPROVED** [review]

File: drivers/gpio/gpio_mcux.c [hunk 1]
```diff
@@ -80,7 +79,12 @@ static int gpio_mcux_configure(const struct device *dev,
 
 	/* Set PCR mux to GPIO for the pin we are configuring */
 	mask |= PORT_PCR_MUX_MASK;
-	pcr |= PORT_PCR_MUX(kPORT_MuxAsGpio);
+	pcr |= PORT_PCR_MUX(PORT_MUX_GPIO);
+
+#if defined(FSL_FEATURE_PORT_HAS_INPUT_BUFFER) && FSL_FEATURE_PORT_HAS_INPUT_BUFFER
+	/* Enable digital input buffer */
+	pcr |= PORT_PCR_IBE_MASK;
```
Comment (henrikbrixandersen) [inline drivers/gpio/gpio_mcux.c:86]: So this should _always_ be enabled? There are no use cases where it should be left disabled?

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
It seems you are defining `MUX_AS_GPIO` to the exact same value for all SoCs.

What is the purpose then? Is this preparation for adding a new SoC which uses a different definition in the MCUX HAL layer? If so, would a simple `#ifdef` in `gpio_mcux.c` suffice?

File: drivers/gpio/gpio_mcux.c (hunk 1 above)
Comment (mmahadevan108) [reply drivers/gpio/gpio_mcux.c:86]: By default this is disabled.

**Comment by mmahadevan108** [comment]
> It seems you are defining `MUX_AS_GPIO` to the exact same value for all SoCs.
> 
> What is the purpose then? Is this preparation for adding a new SoC which uses a different definition in the MCUX HAL layer? If so, would a simple `#ifdef` in `gpio_mcux.c` suffice?

Yes, this in preparation for SoC's where we see variation in which mode is GPIO. I am not a big fan of SoC #defines in the driver code as these tend to get unmanageable.

**Review by dleach02 — APPROVED** [review]

**Comment by mmahadevan108** [comment]
@henrikbrixandersen, hope that addresses your questions

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**